endpoints/openrtb2/amp_auction_test.go

```diff
@@ -122,357 +122,291 @@ func TestAMPPageInfo(t *testing.T) {
-		return
-	}
-
-	// Assert string `consent` is found in the User.Ext at all
-	assert.NotContains(t, fullMarshaledBidRequest, "consent:"+consentString, "This bid request should not contain a consent string. It will be passed the one in the http.Request endpoint")
-
-	// Assert the last request has a valid User object with a consent string equal to that on the URL query
-	var ue openrtb_ext.ExtUser
-	err = json.Unmarshal(exchange.lastRequest.User.Ext, &ue)
-	assert.NoError(t, err, "Error unmarshalling last processed request")
-
-	// Assert consent string found in `http.Request` was passed correctly to the `User.Ext` object
-	assert.Contains(t, string(request.URL.RawQuery), consentString, "http.Request should come with a consent string in its query")
-	assert.Equal(t, consentString, ue.Consent, "Consent string unsuccessfully passed to bid request through AMP endpoint")
-}
-
-func TestSubstituteRequestConsentWithEndpointConsent(t *testing.T) {
-	// gdpr consent string that will come inside our http.Request query
-	const consentString = "BOa71ZYOa71ZYAbABBENA8-AAAAbN7_______9______9uz_Gv_r_f__33e8_39v_h_7_-___m_-3zV4-_lvR11yPA1OrfIrwFhiAw"
-	const DigiTurstID = "digitrustId"
-
-	// Generate a marshaled openrtb.BidRequest that comes with a gdpr consent string
-	fullMarshaledBidRequest, err := getTestBidRequest(false, false, "some-consent-string", "digitrustId")
-	if err != nil {
-		t.Fatalf("Failed to marshal the complete openrtb.BidRequest object %v", err)
+func TestGDPRConsent(t *testing.T) {
+	validConsent := "BONV8oqONXwgmADACHENAO7pqzAAppY"
+
+	digitrust := &openrtb_ext.ExtUserDigiTrust{
+		ID:   "anyDigitrustID",
+		KeyV: 1,
+		Pref: 0,
+	}
+
+	testCases := []struct {
+		description     string
+		consent         string
+		userExt         *openrtb_ext.ExtUser
+		nilUser         bool
+		expectedUserExt openrtb_ext.ExtUser
+	}{
+		{
+			description: "Nil User",
+			consent:     validConsent,
+			nilUser:     true,
+			expectedUserExt: openrtb_ext.ExtUser{
+				Consent: validConsent,
+			},
+		},
+		{
+			description: "Nil User Ext",
+			consent:     validConsent,
+			userExt:     nil,
+			expectedUserExt: openrtb_ext.ExtUser{
+				Consent: validConsent,
+			},
+		},
+		{
+			description: "Overrides Existing Consent",
+			consent:     validConsent,
+			userExt: &openrtb_ext.ExtUser{
+				Consent: "existingConset",
+			},
+			expectedUserExt: openrtb_ext.ExtUser{
+				Consent: validConsent,
+			},
+		},
+		{
+			description: "Overrides Existing Consent - With Sibling Data",
+			consent:     validConsent,
+			userExt: &openrtb_ext.ExtUser{
+				Consent:   "existingConset",
+				DigiTrust: digitrust,
+			},
+			expectedUserExt: openrtb_ext.ExtUser{
+				Consent:   validConsent,
+				DigiTrust: digitrust,
+			},
+		},
+		{
+			description: "Does Not Override Existing Consent If Empty",
+			consent:     "",
```

Comment:
Should we have a test on the behavior if a new and an existing consent both exist?

Reply:
Not a bad idea for documentation purposes. Added.